pymeasure/instruments/teledyne/teledyneMAUI.py

```diff
@@ -0,0 +1,211 @@
+
+    # noinspection PyIncorrectDocstring
+    def setup(self, **kwargs):
+        """Setup channel. Unspecified settings are not modified.
+
+        Modifying values such as probe attenuation will modify offset, range, etc. Refer to
+        oscilloscope documentation and make multiple consecutive calls to setup() if needed.
+        See property descriptions for more information.
+
+        :param bwlimit:
+        :param coupling:
+        :param display:
+        :param offset:
+        :param probe_attenuation:
+        :param scale:
+        :param trigger_coupling:
+        :param trigger_level:
+        :param trigger_slope:
+        """
+        for key, value in kwargs.items():
+            setattr(self, key, value)
+
+    @property
+    def current_configuration(self):
+        """Get channel configuration as a dict containing the following keys:
+
+        - "channel": channel number (int)
+        - "attenuation": probe attenuation (float)
+        - "bandwidth_limit": bandwidth limiting, parsed for this channel (str)
+        - "coupling": "ac 1M", "dc 1M", "ground" coupling (str)
+        - "offset": vertical offset (float)
+        - "display": currently displayed (bool)
+        - "volts_div": vertical divisions (float)
+        - "trigger_coupling": trigger coupling can be "dc" "ac" "highpass" "lowpass" (str)
+        - "trigger_level": trigger level (float)
+        - "trigger_slope": trigger slope can be "negative" "positive" "window" (str)
+        """
+        ch_setup = {
+            "channel": self.id,
+            "attenuation": self.probe_attenuation,
+            "bandwidth_limit": self.bwlimit[f"C{self.id}"],
+            "coupling": self.coupling,
+            "offset": self.offset,
+            "display": self.display,
+            "volts_div": self.scale,
+            "trigger_coupling": self.trigger_coupling,
+            "trigger_level": self.trigger_level,
+            "trigger_slope": self.trigger_slope
+        }
+        return ch_setup
+
+
+class TeledyneMAUI(TeledyneOscilloscope):
+    """A base class for the MAUI-type of Teledyne oscilloscopes.
+
+    This class is not exactly device specific. Nonetheless, it might already work out of the box and
+    therefore this class not abstract.
+
+    The manual detailing the API is "MAUI Oscilloscopes Remote Control and Automation Manual"
+    (`link`_).
+
+    .. _link: https://cdn.teledynelecroy.com/files/manuals/
+              maui-remote-control-automation_27jul22.pdf
+    """
+
+    ch_1 = Instrument.ChannelCreator(TeledyneMAUIChannel, 1)
+
+    ch_2 = Instrument.ChannelCreator(TeledyneMAUIChannel, 2)
+
+    ch_3 = Instrument.ChannelCreator(TeledyneMAUIChannel, 3)
+
+    ch_4 = Instrument.ChannelCreator(TeledyneMAUIChannel, 4)
+
+    # Change listed values for existing commands:
+    bwlimit_values = TeledyneMAUIChannel.BANDWIDTH_LIMITS
+
+    ###############
+    #   Trigger   #
+    ###############
+
+    @property
+    def trigger(self):
```

Comment:
Would "trigger_setup" or "trigger_config" be better names?

Reply:
Probably, but backwards compatibility again.